clang/test/3C/arrboundsprototypemerge.c

```diff
@@ -0,0 +1,24 @@
+// RUN: rm -rf %t*
+// RUN: 3c -base-dir=%S -alltypes %s -- | FileCheck -match-full-lines %s
+// RUN: 3c -base-dir=%S -alltypes %s -- | %clang -c -f3c-tool -fcheckedc-extension -x c -o %t1.unused -
+
+/*
+Handle function prototype merging in bounds inference.
+https://github.com/correctcomputation/checkedc-clang/issues/522
+*/
+
+#include <string_checked.h>
+
+void fiddle_with_strings(const char *tc) {
+  strdup(tc);
+  strcmp(tc, tc);
+  strcmp(tc, "some really long string");
```

Comment:
I'm not sure how this example works with declaration merging. Why not make a test that has declarations directly in the test? E.g., 
```
extern void foo(int *x : itype(_Array_ptr<int>) count(n), int n);
extern void foo(int *x : itype(_Array_ptr<int>), int n);
```
Then you want to make sure this merges to drop the bounds, right?